components/ciphertool/src/main/java/org/wso2/ciphertool/CipherTool.java

```diff
@@ -58,32 +53,30 @@ public class CipherTool {
     public static void main(String[] args) {
 
         initialize(args);
-        Cipher cipher = KeyStoreUtil.initializeCipher();
-        if (System.getProperty(Constants.CONFIGURE) != null &&
-            System.getProperty(Constants.CONFIGURE).equals(Constants.TRUE)) {
+        CipherMode cipherMode = CipherFactory.createCipher();
+        if (Constants.TRUE.equals(System.getProperty(Constants.CONFIGURE))) {
             File deploymentTomlFile = new File(Utils.getDeploymentFilePath());
             if (deploymentTomlFile.exists()) {
                 Map<String, String> secretMap = Utils.getSecreteFromConfiguration(Utils.getDeploymentFilePath());
                 for (Map.Entry<String, String> entry : secretMap.entrySet()) {
                     String key = entry.getKey();
                     String value = Utils.getUnEncryptedValue(entry.getValue());
                     if (StringUtils.isNotEmpty(value)) {
-                        String encryptedValue = Utils.doEncryption(cipher, value);
+                        String encryptedValue = cipherMode.doEncryption(value);
                         secretMap.replace(key, encryptedValue);
                     }
                 }
                 updateDeploymentConfigurationWithEncryptedKeys(secretMap);
             } else {
                 loadXpathValuesAndPasswordDetails();
                 secureVaultConfigTokens();
-                encryptCipherTextFile(cipher);
+                encryptCipherTextFile(cipherMode);

```

Comment:
Was this encryption of xml files verified?

Reply:
encryption of xml files is not used in Identity Server since we use deployment.toml file.